src/book/entities/Account.ts

```diff
@@ -309,3 +310,38 @@ function CheckIECommodity(validationOptions?: v.ValidationOptions) {
     });
   };
 }
+
+/**
+ * Checks if an account can be a placeholder. Accounts can be Placeholders
+ * if they don't have any transactions associated.
+ */
+function CheckPlaceholder(validationOptions?: v.ValidationOptions) {
+  return function f(object: Account, propertyName: string) {
+    v.registerDecorator({
+      name: 'checkPlaceholder',
+      target: object.constructor,
+      propertyName,
+      options: validationOptions,
+      validator: {
+        async validate(_: string, args: v.ValidationArguments) {
```

Comment:
first async validator I think. Awesome that it works, this will enable to create better validation in the future